### DISCUSSION TITLE
Calling `delete(id)` on a record returns with a 404

### DISCUSSION THREAD

**Discussion by aashishvasu** [pocketbase/pocketbase#2351] [Q&A]
I am calling delete on a record after successfully finding its id from the record entry. Console logging the record says that it is valid. But when the delete function is called, I am getting this error
```
url: 'http://127.0.0.1:8090/api/collections/actors/records/7ltarn6v1e181qd',
  status: 404,
  response: {
    code: 404,
    message: "The requested resource wasn't found.",
    data: {}
  },
  isAbort: false,
  originalError: null
```

I have verified that the record exists, and even manually copying the ID from the admin panel into the delete function parameter returns the same error.
The same logic works on another collection, so that has me really stumped (collection `projects` instead of `actors` seems to work). Both collections have the same API rules for deletion, and are called under similar circumstances.
code for calling delete:
```javascript
// Works
await locals.pb.collection('projects').delete(inData.id);

// Does not work
await locals.pb.collection('actors').delete(inData.id);
```
And finally, here is the log from the admin panel:
![image](https://user-images.githubusercontent.com/6966900/233918706-0ec0abeb-3a75-435e-8afe-a39b2c347a2d.png)

What other things can I check, apart from the usual checking if the collection/records exist etc.? Any help on this would be great!

**Comment by ganigeorgiev** [comment]
Make sure that your client request satisfies the Delete API rule of the collection.

If that's not the case, please export your collections configuration (Admin UI > Settings > Export collections) and I'll try to investigate it.

**Comment by inactdev** [comment]
@aashishvasu I was having a similar issue until I realized that I have the API rule setup wrong. On my Entries collection, ensure I was only allowing the user who created the record to Update or Destroy it, I initially had my API rules setup like this:

![Screen Shot 2023-09-02 at 5 58 14 PM](https://github.com/pocketbase/pocketbase/assets/10625312/12502695-b396-4a57-83b5-27bd7bf05bfb)

`@request.auth` here is the authenticated model, which is the User. So this setup was comparing the id of the Entry with the authenticated model id, aka the User id. My model references a User with a column called `user`. So, for me, I just had to change the API rule to:

![Screen Shot 2023-09-02 at 6 03 35 PM](https://github.com/pocketbase/pocketbase/assets/10625312/421d4e86-e8f8-43d2-9094-7e6c3406e5cc)

And all was well !! 🎉  (I know it's a little confusing that the User id is in a column called `user` and not `user_id`, but this seems to be a Pocketbase convention).

I will say that the error message in the Request Log isn't very helpful 😕 I understand it's best to avoid revealing any info to a potentially malicious actor, but maybe somewhere in the Logs UI a better message could be provided ? 

Something like `id did not match @request.auth.id` would've been SUPER helpful 😄 

BTW @ganigeorgiev, every single issue that I've found on github has been answered by you !! Not sure if you're the only one working on this but you're a saint for answering all these questions !! Much appreciated !! 🙏
